game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -885,40 +888,61 @@ private void determineUnitsToAttackWith(
       }
 
       // Check to see if any territories can be bombed
+      final Map<Territory, BombedTerritoryData> bombedTerritoryDataMap = new HashMap<>();
       final Map<Unit, Set<Territory>> bomberMoveMap =
           territoryManager.getAttackOptions().getBomberMoveMap();
       for (final Unit unit : bomberMoveMap.keySet()) {
         if (alreadyAttackedWithUnits.contains(unit)) {
           continue;
         }
+        final Predicate<Unit> bombingTargetMatch =
+            Matches.unitCanProduceUnitsAndCanBeDamaged()
+                .and(Matches.unitIsLegalBombingTargetBy(unit));
         Optional<Territory> maxBombingTerritory = Optional.empty();
         int maxBombingScore = MIN_BOMBING_SCORE;
         for (final Territory t : bomberMoveMap.get(unit)) {
-          final boolean canBeBombedByThisUnit =
-              t.getUnitCollection()
-                  .anyMatch(
-                      Matches.unitCanProduceUnitsAndCanBeDamaged()
-                          .and(Matches.unitIsLegalBombingTargetBy(unit)));
+          final boolean canBeBombedByThisUnit = t.getUnitCollection().anyMatch(bombingTargetMatch);
           final boolean canCreateAirBattle =
               Properties.getRaidsMayBePreceededByAirBattles(data.getProperties())
                   && AirBattle.territoryCouldPossiblyHaveAirBattleDefenders(t, player, data, true);
           if (canBeBombedByThisUnit
               && !canCreateAirBattle
               && canAirSafelyLandAfterAttack(unit, t)) {
-            final int noAaBombingDefense =
-                t.getUnitCollection().anyMatch(Matches.unitIsAaForBombingThisUnitOnly()) ? 0 : 1;
+            // get territory data that is independent of the attack unit
+            final BombedTerritoryData bombedTerritoryData =
+                bombedTerritoryDataMap.computeIfAbsent(
+                    t,
+                    territory -> {
+                      int noAaBombingDefense = 1;
+                      // minimum damage to allow bombing = max. damage a defense unit can take
+                      int minDamageNeeded = 0;
+                      final List<Unit> potentialBombingTargetUnits =
+                          t.getUnitCollection().getMatches(bombingTargetMatch);
+                      for (final Unit targetUnit : potentialBombingTargetUnits) {
+                        minDamageNeeded = Math.max(minDamageNeeded, targetUnit.getUnitDamage());
+                        if (Matches.unitIsAaForBombingThisUnitOnly().test(targetUnit)) {
```

Comment:
Do we have logic like this somewhere already in the codebase in a utility class? Ideally, we could reuse logic, so if there are rules updates or other changes, it stays consistent.